dbms/src/Flash/Pipeline/Schedule/TaskThreadPool.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Exception.h>
+#include <Common/Stopwatch.h>
+#include <Common/setThreadName.h>
+#include <Flash/Pipeline/Schedule/Task/TaskHelper.h>
+#include <Flash/Pipeline/Schedule/TaskScheduler.h>
+#include <Flash/Pipeline/Schedule/TaskThreadPool.h>
+#include <common/likely.h>
+#include <common/logger_useful.h>
+
+#include <magic_enum.hpp>
+
+namespace DB
+{
+TaskThreadPool::TaskThreadPool(TaskScheduler & scheduler_, size_t thread_num)
+    : scheduler(scheduler_)
+{
+    RUNTIME_CHECK(thread_num > 0);
+    threads.reserve(thread_num);
+    for (size_t i = 0; i < thread_num; ++i)
+        threads.emplace_back(&TaskThreadPool::loop, this);
+}
+
+void TaskThreadPool::close()
+{
+    task_queue->close();
+}
+
+void TaskThreadPool::waitForStop()
+{
+    for (auto & thread : threads)
+        thread.join();
+    LOG_INFO(logger, "task thread pool is stopped");
+}
+
+void TaskThreadPool::loop() noexcept
+{
+    setThreadName("TaskThreadPool");
```

Comment:
Should we add the thread number into the thread name? This could be more verbose for troubleshooting.

Reply:
ok, added.